app/src/main/java/beepbeep/pixels/feed/FeedInteractor.kt

```diff
@@ -0,0 +1,30 @@
+package beepbeep.pixels.feed
+
+import beepbeep.pixels.feed.list.FeedListViewModel
+import io.reactivex.Observable
+
+class FeedInteractor(input: FeedContract.Input, source: FeedDataSource) : FeedContract.Output {
+
+    private var viewModels = Observable.empty<FeedViewModel>()
+
+    override val loadings: Observable<Boolean> by lazy { viewModels.map(FeedViewModel::isLoading) }
+
+    override val items: Observable<List<FeedListViewModel>> by lazy { viewModels.map(FeedViewModel::items) }
+
+    init {
+        val loadAction = source.loadData()
+                .map { SetAction(it) as FeedAction }
+                .startWith(LoadAction)
+
+        val refreshAction = input.onRefresh
+                .flatMap {
+                    Observable.just(LoadAction as FeedAction).concatWith(source.loadData().map { SetAction(it) })
+                }
+
+        viewModels = Observable.mergeArray(loadAction, refreshAction)
+                .scan(FeedViewModel(), ::reduceWithAction)
+                .skip(1)
```

Comment:
[question]
`skip(1)` is for skipping the initial state of `ViewModel`?
So it will be `startWith(LoadAction)`, it's like saying, at the beginning it will be loading?

Reply:
.scan(FeedViewModel(), ::reduceWithAction)